bin/monitor

```diff
@@ -2,62 +2,96 @@
 # frozen_string_literal: true
 
 require_relative "../loader"
+clover_freeze
 
 MONITORABLE_RESOURCE_TYPES = [PostgresServer, MinioServer, GithubRunner]
+resources = {}
+mutex = Mutex.new
 
-sessions = {}
-ssh_threads = {}
-pulse_threads = {}
+producer = Thread.new do
+  loop do
+    mutex.synchronize do
+      MONITORABLE_RESOURCE_TYPES.flat_map { _1.all }.each do |r|
+        resources[r.id] ||= {
+          model: r,
+          session: r.reload.init_health_monitor_session,
```

Comment:
The resource might be deleted between the time we fetch all resources and process the current one. If the resource is deleted `init_health_monitor_session` can fail due to different reasons, reload reconciles all of them to `Sequel::NoExistingObject`, which will be caught below and ignored.

Reply:
But there's no barrier preventing that situation occurring again, right? Like, at no point there's any monotonic logic that prevents the record from disappearing at any time here.

Reply:
I didn't understand your comment.

This check is for the case where the resource is deleted between the time it is materialized by the `.each` call above and the time it is iterated. producer loop runs in every 5 minutes. If the resource is deleted and ignored here, in the next iteration, it wouldn't materialized at all.

Resource might be deleted after producer completes its job, for that case there is another `Sequel::NoExistingObject` rescue below.

Reply:
I guess my point is, the `.reload` does not interlock with anything, the moment is has run, the record can then be deleted. So what purpose does it serve? Downstream code has to be robust to a disappearing record anyway.

Reply:
I understand your point now. To fix that, we need to catch all exceptions and then we can remove the reload call. I did that, but now we have bunch of `rescue => ex` statements in the monitor code, which I'm not very happy with.

I also didn't removed the second `reload` call, because I think that still serves its purpose. If the resource is deleted between the reload and process, its session will be deleted and the reload call will be executed again in next loop, eventually raising `Sequel::NoExistingObject` exception.